File: Backend/library-management-system/src/main/java/com/srijon/library/service/MemberService.java
package com.srijon.library.service;

import com.srijon.library.dto.Member.MemberRequestDto;
import com.srijon.library.dto.Member.MemberResponseDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface MemberService {

    //Create a member
    MemberResponseDto addMember(MemberRequestDto memberRequestDto);

    //Get all the members
    Page<MemberResponseDto> getAllMembers(Pageable pageable);

    //Get members by id
    MemberResponseDto getMemberById(Long id);

    //Update a member
    MemberResponseDto updateMember(Long id, MemberRequestDto memberRequestDto);

    //Delete a member
    void deleteMember(Long id);

    //Search a member
    Page<MemberResponseDto> searchMember(
            String query, Pageable pageable
    );
}

File: Frontend/library-management-frontend/src/app/services/MemberService.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Member } from '../models/member';

@Injectable({
  providedIn: 'root',
})
export class MemberService {
  private http = inject(HttpClient);

  private apiUrl = 'http://localhost:8080/api/members';

  getAllMembers(page: number, size: number): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}?page=${page}&size=${size}`);
  }

  getMemberById(id: number): Observable<Member> {
    return this.http.get<Member>(`${this.apiUrl}/${id}`);
  }

  createMember(member: Member): Observable<Member> {
    return this.http.post<Member>(this.apiUrl, member);
  }

  updateMember(id: number, member: any): Observable<Member> {
    return this.http.put<Member>(`${this.apiUrl}/${id}`, member);
  }

  deleteMember(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  searchMembers(keyword: string): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/search?query=${keyword}`);
  }
}
